tests/incident/test_views.py

```diff
@@ -24,101 +40,318 @@ def tearDown(self):
         # Start with no incidents or tags
         self.assertFalse(Incident.objects.exists())
         self.assertFalse(Tag.objects.exists())
+        self.assertFalse(IncidentTagRelation.objects.exists())
         # Minimal data to post that has tags
         data = {
             "start_time": "2021-08-04T09:13:55.908Z",
             "end_time": "2021-08-04T09:13:55.908Z",
-            "description": "string",
+            "description": "incident",
             "level": 1,
             "tags": [{"tag": "a=b"}],
         }
-        response = self.client.post("/api/v1/incidents/", data, format="json")
+        response = self.client.post(path="/api/v1/incidents/", data=data, format="json")
         self.assertEqual(response.status_code, 201)  # Created
         # Check that we have made the correct Incident
         self.assertEqual(response.data["description"], data["description"])
         self.assertTrue(Incident.objects.exists())
-        obj = Incident.objects.get()
-        self.assertEqual(obj.description, data["description"])
+        incident = Incident.objects.get()
+        incident_tags = [relation.tag for relation in IncidentTagRelation.objects.filter(incident=incident)]
+        self.assertEqual(incident.description, data["description"])
         # Check that we have made the correct Tag
+        self.assertTrue(IncidentTagRelation.objects.exists())
         self.assertTrue(Tag.objects.exists())
         tag = Tag.objects.get()
-        self.assertEqual(obj.tags, [tag])
+        self.assertEqual(incident_tags, [tag])
         self.assertEqual(str(tag), data["tags"][0]["tag"])
 
+    def test_get_incident_by_pk_returns_correct_incident(self):
+        incident_pk = self.add_incident()
+        response = self.client.get(path=f"/api/v1/incidents/{incident_pk}/")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["pk"], incident_pk)
+
     def test_incident_can_properly_change_level(self):
-        incident_pk = self.add_incident("incident1")
+        incident_pk = self.add_incident()
         incident_path = reverse("v1:incident:incident-detail", args=[incident_pk])
         response = self.client.put(
-            incident_path,
-            {
+            path=incident_path,
+            data={
                 "tags": [{"tag": "a=b"}],
                 "level": 2,
             },
         )
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         self.assertEqual(Incident.objects.get(pk=incident_pk).level, 2)
 
+    def test_incident_acks_returns_correct_acks(self):
+        incident_pk = self.add_incident()
+        ack_pk = self.add_acknowledgement(incident_pk=incident_pk)
+        response = self.client.get(path=f"/api/v1/incidents/{incident_pk}/acks/")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data[0]["pk"], ack_pk)
+        self.assertEqual(response.data[0]["event"]["type"]["value"], "ACK")
```

Comment:
(continuation of previous comment) here use the fields from the ack object directly to get incident pk
```suggestion
    def test_incident_acks_returns_correct_acks(self):
        ack = self.add_acknowledgement()
        response = self.client.get(path=f"/api/v1/incidents/{ack.event.incident.pk}/acks/")
        self.assertEqual(response.status_code, 200)
        self.assertEqual(response.data[0]["pk"], ack.pk)
        self.assertEqual(response.data[0]["event"]["type"]["value"], "ACK")
```